packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -110,8 +117,12 @@ function getWrappedShadowRootListener(listener: EventListener): WrappedListener
                 shouldInvoke = shouldInvokeListener(event, actualTarget, currentTarget);
             }
 
-            if (shouldInvoke) {
-                listener.call(currentTarget, event);
+            try {
+                if (shouldInvoke) {
+                    listener.call(currentTarget, event);
+                }
+            } finally {
+                endHandledByWrappedListener(event);
             }
```

Comment:
nit: To keep the relevant parts of the logic contained in one block.
```suggestion
            if (shouldInvoke) {
                startHandledByWrappedListener(event);
                try {
                    listener.call(currentTarget, event);
                } finally {
                    endHandledByWrappedListener(event);
                }
            }
```